### PR TITLE
Add pyqgis cookbook code about selecting a feature with the maptool S…

### PR DESCRIPTION
**Pull request by marcu**
…election of a feature using QgsMapToolIdentifyFeature

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 1]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
 
 .. index:: Map canvas; Custom map tools
 
+
+Selection of a feature using QgsMapToolIdentifyFeature
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:299]: Does it **select**?

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 2]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+------------------------------------------------------
+
+You can use the map tool QgsMapToolIdentifyFeature for asking to the
+user to select a feature that wiil be send to a callback function.
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:303]:
```suggestion
You can use the map tool :class:`QgsMapToolIdentifyFeature <qgis.gui.QgsMapToolIdentifyFeature>`
for asking to the user to select a feature that will be sent to a callback function.
```
the :class: tag turns this into a hyperlink pointing to the corresponding section in the pyqgis doc.
I'm not sure I understand the end of the sentence.

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 3]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+
+.. testcode:: canvas
+
+  def callback(feature):
+    """Code called when the feature is selected by the user"""
+    print(f"You clicked on feature {feature.id()}")
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:309]:
```suggestion
    print("You clicked on feature {}".format(feature.id()))
```
?

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 4]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+
+  feature_identifier = QgsMapToolIdentifyFeature(canvas)
+
+  # indicates the layer on which the selection wiil be done
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:313]:
```suggestion
  # indicates the layer on which the selection will be done
```

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 5]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+  feature_identifier.setLayer(vlayer)
+
+  # use the callback as a slot triggered when the user identify a feature
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:316]:
```suggestion
  # use the callback as a slot triggered when the user identifies a feature
```

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 6]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+  feature_identifier.featureIdentified.connect(callback)
+
+  # activation of hte map tool
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:319]:
```suggestion
  # activation of the map tool
```

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 7]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+  canvas.setMapTool(feature_identifier)
+
+
+
 Writing Custom Map Tools
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:324]:
```suggestion
.. index:: Map canvas; Custom map tools

Writing Custom Map Tools
```

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 8]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
 
 .. index:: Map canvas; Custom map tools
```
Comment (DelazJ) [inline docs/pyqgis_developer_cookbook/canvas.rst:297]: this line refers to the next section then should probably be closer (moved back)
```suggestion
```

**Review by DelazJ — COMMENTED** [review]
Thanks @marcu
I left some styling/typo feedback and question. I'll let more pyqgis versed people review the rest.

File: docs/pyqgis_developer_cookbook/canvas.rst [hunk 9]
```diff
@@ -295,6 +295,32 @@ selected layer on the newly created canvas
...
+
+You can use the map tool QgsMapToolIdentifyFeature for asking to the
+user to select a feature that wiil be send to a callback function.
+
+.. testcode:: canvas
+
+  def callback(feature):
+    """Code called when the feature is selected by the user"""
+    print(f"You clicked on feature {feature.id()}")
+
```
Comment (ghtmtt) [inline docs/pyqgis_developer_cookbook/canvas.rst:310]: I'll add the creation of the `canvas` instance:

canvas = iface.mapCanvas()

**Review by ghtmtt — COMMENTED** [review]
LGTM! Just a small suggestion.

**Comment by marcu** [comment]
Hello,

Thanks for the feedback and the correction. I will fix it tomorrow.

What should I do about the "Backport to LTR documentation is required" ?

**Comment by DelazJ** [comment]
> What should I do about the "Backport to LTR documentation is required" ?

No worries. We (actually the backport bot) take care of that...

**Comment by marcu** [comment]
So is it ok for merging ?

**Pull request merged**